src/app/dataset/actionbar/actionbar.component.ts

```diff
@@ -142,6 +144,16 @@ export class ActionbarComponent {
     );
   }
 
+  /** canCancelWorkflow
+  /*  calculate if the workflow can be cancelled
+  */
+  canCancelWorkflow(): boolean {
+    return (
+      [`${PluginType.PUBLISH}`, `${PluginType.DEPUBLISH}`].indexOf(`${this.currentPluginName}`) ==
+        -1 && !this.isCompleted
+    );
```

Comment:
Seems like duplicate code - you added this check twice. Is there a possibility to have one source of truth somewhere?

Reply:
It was similar - so making the helper wasn't painless.